test/unit/style/style.test.js

```diff
@@ -1006,6 +1006,28 @@ test('Style#addLayer', (t) => {
         });
     });
 
+    t.test('warns if before layer does not exist', (t) => {
+        const style = new Style(new StubMap());
+        style.loadJSON(createStyleJSON({
+            layers: [{
+                id: 'a',
+                type: 'background'
+            }, {
+                id: 'b',
+                type: 'background'
+            }]
+        }));
+        const layer = {id: 'c', type: 'background'};
+        t.stub(console, 'error');
```

Comment:
Bind a handler for the `error` event rather than stubbing `console.error` (see the subsequent test for example).